src/metadata.hh

```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <QStringView>
+#include <optional>
+
+/**
+ * Represent the metadata.toml beside the dictionary files
```

Comment:
a namespace?

Reply:
fixed 

Not sure about the naming of this struct. `Metadata::result` feels OK. :sweat_smile:

Reply:
Misc  abbrevation of miscellaneous?

Reply:
Misc feels like "extra things that may or may not be interesting"

This struct is the core structure that represents the result of parsing the metadata.toml.

Naming is too hard :sweat_smile: